src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -487,6 +491,21 @@ private void ProcessSettingsFrame(FrameHeader frameHeader)
 
                             // We don't actually store this value; we always send frames of the minimum size (16K).
                             break;
+                        case SettingId.HeaderTableSize:
+                            if (settingValue < 0)
+                            {
+                                throw new Http2ConnectionException(Http2ProtocolErrorCode.ProtocolError);
+                            }
+                            try
+                            {
+                                _hpackEncoder.SetDynamicHeaderTableSize(settingValue);
+                            }
+                            catch (HPackEncodingException)
+                            {
+                                throw new Http2ConnectionException(Http2ProtocolErrorCode.CompressionError);
```

Comment:
Question: Http2ConnectionException contains compression error which is intended for encoding and decoding headers. However, we use different exception type in encoder and decoder